src/HeapedVectors.jl

```diff
@@ -0,0 +1,156 @@
+__precompile__()
+
+struct HeapedVector{T, F<:Function}  
+    data::Vector{T}
+    by::F
+    function HeapedVector(v::Vector{T}, by::F) where {T, F}
+        new{T, F}(heaping(v, by), by)
+    end
+end
+      
+HeapedVector(data::Vector{T}) where {T} = HeapedVector(data, identity)  
+
+       
+function heaping(v, by) 
+    ar = []
+    for i = 1:length(v) 
+        insert!(ar, i, v[i])
+        floatup(length(ar), ar, by)
+    end
+    return ar
+end
+
+function floatup(index, ar, by)	
+    par = convert(Int, floor(index/2))
+    if index <= 1
+        return ar 
+    end
+    if by(ar[index]) < by(ar[par])
+        temp = ar[par]
+        ar[par] = ar[index]
+        ar[index] = temp
+    end
+    floatup(par, ar, by)
+end
+
+
+function push!(v::HeapedVector{T}, x::T) where {T}
+    insert!(v.data, length(v.data)+1, x)
+    floatup(length(v.data), v.data, v.by)
+    return v
+end
+
+
+isempty(v::HeapedVector) = isempty(v.data)
+
+
+function popfirst!(v::HeapedVector{T}) where {T}
+    if length(v.data) == 0
+         return 
+    end
+	
+    if length(v.data) > 2
+        temp = v.data[length(v.data)]
+    	v.data[length(v.data)] = v.data[1]
+    	v.data[1] = temp
+    	Min = pop!(v.data)
```

Comment:
Don't use capital letters for variables. Use same trick again.

Reply:
ok I am changing "Min" to "minm"